### PR TITLE
Support Java 14

### PR DESCRIPTION
**Pull request by artembilan**
* Provide changes to avoid deprecated Java API
and have a compatibility back to Java 8
* Change affected test classes to JUnit 5 whenever it is possible
* Ignore/Disable some TCP/IP tests which don't pass on Java 14

### CODE REVIEW

**Comment by artembilan** [comment]
@garyrussell ,

I need your help with TCP/IP.

It looks like there are some changes in the UDP multicast with their `NetworkInterface` and `joinGroup()`.
Plus there is no `open` field in the `SocketChannel` any more.
See `@EnabledOnJre(JRE.JAVA_8)` tests.
And `SocketSupportTests.testNioClientAndServerSSLDifferentContexts()` fails for no reason for me.

Thanks

File: spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/SocketSupportTests.java [hunk 1]
```diff
@@ -518,7 +518,7 @@ public void testNioClientAndServerSSL() throws Exception {
 	}
 
 	@Test
-	@EnabledOnJre(JRE.JAVA_8)
+	@EnabledOnJre(JRE.JAVA_8) // Need to wait for write complete on Java 14
```
Comment (artembilan) [inline spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/SocketSupportTests.java]: @garyrussell , what does it mean?
Is there a way to do something similar what you have with that `if (System.getProperty("java.version").startsWith("1.8")) {` ?

Comment (garyrussell) [reply spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/SocketSupportTests.java]: I fixed it in the next commit.

**Review by garyrussell — APPROVED** [review]

**Comment by artembilan** [comment]
Gary, I'm OK to merge this, but may be you have some idea about that deprecated `MulticastSocket.joinGroup()`?
Or you just can't investigate it altogether because your Mac doesn't support multicast?
Thanks

**Comment by garyrussell** [comment]
Right; I need to test in a Linux VM. 

Please open an issue and assign to me so I don't forget about it.

**Comment by garyrussell** [comment]
Should we add a build plan?

**Comment by artembilan** [comment]
> Should we add a build plan?

Sure! On it...

**Pull request merged**